src/context/layers/groundstation.ts

```diff
@@ -15,25 +16,37 @@ declare module 'geojson' {
 export type GroundstationLayerData = {
   lat: number;
   lon: number;
+  date: string; // yyyy-mm-dd
   [key: string]: any;
 }[];
 
-export const fetchGroundstationData: LazyLoader<GroundstationLayerProps> = () => async (
-  params: LayerDataParams<GroundstationLayerProps>,
-) => {
+export const fetchGroundstationData: LazyLoader<GroundstationLayerProps> = () => async ({
+  date,
+  layer,
+}: LayerDataParams<GroundstationLayerProps>) => {
   // This function fetches groundstation data from the API.
   // If this endpoint is not available or we run into an error,
   // we should get the data from the local public file in layer.fallbackData
 
-  const { layer } = params;
+  const formattedDate = date && moment(date).format('YYYY-MM-DD');
 
+  const dateQuery = `?beginDateTime=${
+    formattedDate || layer.beginDate
+  }&endDateTime=${formattedDate || layer.endDate}`;
   let data;
   try {
     // eslint-disable-next-line fp/no-mutation
-    data = await (await fetch(layer.data, { mode: 'cors' })).json();
+    data = (await (
+      await fetch(layer.data + dateQuery, { mode: 'cors' })
+    ).json()) as GroundstationLayerData;
   } catch (ignored) {
     // eslint-disable-next-line fp/no-mutation
-    data = await (await fetch(layer.fallbackData || '')).json();
+    data = ((await (
+      await fetch(layer.fallbackData || '')
+    ).json()) as GroundstationLayerData).filter(obj => {
+      // console.log(obj.date, formattedDate, moment(obj.date).valueOf(), date);
+      return moment(obj.date).valueOf() === moment(formattedDate).valueOf();
```

Comment:
wouldn't it be easier to check for `string` equality here?

Reply:
sometimes the api returns `2020-4-01` instead of `2020-04-01`. This helps make sure that despite the inconsistencies in formatting, the date is always recognized correctly. I'll add a comment here to clarify this :+1: